libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -629,8 +629,46 @@ bool ContractCompiler::visit(InlineAssembly const& _inlineAssembly)
 			}
 			else if (auto variable = dynamic_cast<VariableDeclaration const*>(decl))
 			{
-				solAssert(!variable->isConstant(), "");
-				if (m_context.isStateVariable(decl))
+				if (variable->isConstant())
+				{
+					u256 value;
+					if (variable->value()->annotation().type->category() == Type::Category::RationalNumber)
+					{
+						value = dynamic_cast<RationalNumberType const&>(*variable->value()->annotation().type).literalValue(nullptr);
+						if (FixedBytesType const* bytesType = dynamic_cast<FixedBytesType const*>(variable->type()))
+							value = value << (256 - 8 * bytesType->numBytes());
+						else
+							solAssert(variable->type()->category() == Type::Category::Integer, "");
```

Comment:
What about fixed point?

Reply:
Are you saying fixed points should be implemented here or they should be prevented from passing through this code?

Reply:
My question is rather what happens then. Does it reach an assert(false)?

Reply:
yes, it asserts here.